pkg/validate/interface.go

```diff
@@ -0,0 +1,44 @@
+package validate
+
+// Validator is an interface for implementing a validator of a single
+// Kubernetes object type. Ideally each Validator will check one aspect of
+// an object, or perform several steps that have a common theme or goal.
+type Validator interface {
+	// Validate should run validation logic on an arbitrary object, and return
+	// an error if the object does not pass validation.
+	Validate() error
```

Comment:
Returning `[]error` here might be useful, or defining an error return type with appropriate fields.

Reply:
Sure, I will export it and use it as a return type. Instead of `panic()`ing we can add an `Err error` field that the caller can check `== nil`. However I don't expect many, if any, non-validation-related errors to be encountered while `Validate()`ing, like those you mentioned, because functions that generate those errors are usually called in preceding steps.

Reply:
Great, thanks!